src/integrationtests/java/com/aws/greengrass/integrationtests/SyncTest.java

```diff
@@ -433,21 +428,27 @@ void GIVEN_synced_shadow_WHEN_local_update_THEN_cloud_updates(Class<?extends Bas
             InterruptedException, IoTDataPlaneClientCreationException {
         ignoreExceptionOfType(context, ResourceNotFoundException.class);
         ignoreExceptionOfType(context, InterruptedException.class);
-
+        CountDownLatch cdl = new CountDownLatch(1);
         // no shadow exists in cloud
         when(iotDataPlaneClientFactory.getIotDataPlaneClient().getThingShadow(any(GetThingShadowRequest.class)))
                 .thenThrow(ResourceNotFoundException.class);
 
         // mock response to update cloud
         when(mockUpdateThingShadowResponse.payload()).thenReturn(SdkBytes.fromString("{\"version\": 1}", UTF_8));
         when(iotDataPlaneClientFactory.getIotDataPlaneClient().updateThingShadow(cloudUpdateThingShadowRequestCaptor.capture()))
-                .thenReturn(mockUpdateThingShadowResponse);
+                .thenAnswer((ignored)->{
+                    cdl.countDown();
+                    return mockUpdateThingShadowResponse;
+                });
 
         startNucleusWithConfig(NucleusLaunchUtilsConfig.builder()
                 .configFile(getSyncConfigFile(clazz))
                 .syncClazz(clazz)
                 .mockCloud(true)
                 .build());
+        // Let the initial FullShadowSyncRequests finish before the local update request. Otherwise, the local shadow
+        // will be delete during the FullShadowSyncRequest.
+        verify(syncQueue, after(10000).atMost(4)).put(any(FullShadowSyncRequest.class));
```

Comment:
The test config file used for periodic sync has 4 shadows (and realtime sync as 1 shadow to sync) in the shadow manager component configuration. So, when the sync starts up for the first time, full shadow sync is performed on all 4 shadows.  Since checks are same for both the strategies, verifying 4 calls atmost.